### DISCUSSION TITLE
Log/Stats for monitored website

### DISCUSSION THREAD

**Discussion by gert-creatic** [spatie/laravel-uptime-monitor#336] [Ideas]
Is it possible to add a downtime log for each monitor in any way (i would like to add a chart showing the uptime for the last month for example).

**Comment by milosz-rycko-bozenski** [comment]
@gert-creatic Disclaimer: I found this package today so I'm not sure that this is correct way to achieve this, but at the moment I think that the best way is described below.

in `config/uptime-monitor.php` there is line `'response_checker' => Spatie\UptimeMonitor\Helpers\UptimeResponseCheckers\LookForStringChecker::class,`. `LookForStringChecker::class` implements interface from `UptimeResponseChecker`. So you can crete two methods: `isValidResponse` and `getFailureReason` and implement algorithm (pseudo code) below:

```
for every monitor {
   current_status = get monitor status
   previous_status = select from database last entry for this monitor

   if there is no previous_status { add new entry: monitor_id, status=current_status, from=current_timestamp, to="" }
   else {
      if current_status == UP && previous_status == UP: do nothing
      if current_status == UP && previous_status == DOWN: set to=current_timestamp for previous entry and add new entry: monitor_id, status=UP, from=current_timestamp, to=""
      if current_status == DOWN && previous_status == DOWN: do nothing
      if current_status == DOWN && previous_status == UP: set to=current_timestamp for previous entry and add new entry: monitor_id, status=DOWN, from=current_timestamp, to="" 
   {
}

```

if you code this, then you will easily calculate stats for particular monitor.

Also you need to create migration and create custom database table to store data from algorithm above.

I think that I will implement this today or tomorrow and I will post here. Maybe my code will be used ;-)

**Reply by jappi00** [reply]
This is certainly a way to do it. But I would suggest using the implemented Events for this purpose. What do you think?
